### ISSUE TITLE
[discovery] Implement composite design tokens 

### ISSUE THREAD

**Issue by adamstankiewicz** [openedx/paragon#2025]
As a follow-up to https://github.com/openedx/paragon/pull/1992 (https://github.com/openedx/paragon/issues/1947), we would like to migrate to using composite tokens similar to what's proposed in the W3C spec for the following tokens types:

* Shadow
* Transition
* Typography
* Border

By using composite tokens, each component of a style property is explicitly defined as its own attribute as opposed to assuming the value to be a platform-specific implementation (e.g., CSS).

`style-dictionary` does support composite tokens but its default transforms/filters do no handle them already, so we may need to write custom transformers. There is a collection of useful transformers we might be able to use as a starting point. Examples:
* [Transforming composite shadow tokens to CSS](https://github.com/lukasoppermann/style-dictionary-utils/blob/main/src/transformer/shadow-css.ts)
* [Transforming composite typography tokens to CSS](https://github.com/lukasoppermann/style-dictionary-utils/blob/main/src/transformer/font-css.ts)

**Additional Resources**
* https://design-tokens.github.io/community-group/format/#composite-design-token
* https://design-tokens.github.io/community-group/format/#shadow
* https://design-tokens.github.io/community-group/format/#transition
* https://design-tokens.github.io/community-group/format/#typography
* https://design-tokens.github.io/community-group/format/#border

**Comment by peterkulko** [comment]
I think we've already managed to add composite design tokens as part of our work on this PR: https://github.com/openedx/paragon/pull/3222

**Comment by brian-smith-tcril** [comment]
@PKulkoRaccoonGang I agree that #3222 addressed this. I think the only remaining gaps in composite token implementation are being tracked by https://github.com/openedx/paragon/issues/1938, so I'm going to close this one.
